src/components/Comment-form.jsx

```diff
@@ -1,11 +1,66 @@
-import React from "react";
+import { React, useState } from "react";
 
-const CommentForm = () => {
-  return (
-    <form>
-      <label></label>
-    </form>
-  );
+const CommentForm = ({
+  comment,
+  setComments,
+  article_id,
+  setCommentChange,
+  postComment,
+  username,
+}) => {
+  const [newComment, setNewComment] = useState("Enter comment here...");
+  const [isValid, setIsValid] = useState(true);
+  if (comment) {
+    return (
+      <form
+        className="commentForm"
+        onSubmit={(e) => {
+          e.preventDefault();
+          if (newComment === "") {
+            setIsValid(false);
+            setNewComment("Please enter comment");
+          } else if (newComment === "Enter comment here...") {
+            setIsValid(false);
+          } else if (newComment === "Please enter comment") {
+            setIsValid(false);
+          } else if (newComment !== "") {
+            setComments((currComments) => {
+              let comm = {
+                votes: 0,
+                author: username,
+                body: newComment,
+                created_at: Date.now(),
+                comment_id: "TBC",
+              };
+              return [comm, ...currComments];
+            });
+            postComment(newComment, username, article_id);
+            setCommentChange(true);
+            setNewComment("");
+          }
```

Comment:
This optimistic rendering for comments isn't quite doing what you hope. It is adding the comment into the array, but because you are then calling `setCommentChange` it is triggering the `useEffect` inside the `Article-comments` component, which forces the comments to be fetched again and then re-rendered. This second re-render probably isn't necessary - not the end of the world right now, but something to think about!